pkg_resources/__init__.py

```diff
@@ -491,19 +496,6 @@ def compatible_platforms(provided, required):
     return False
 
 
-def run_script(dist_spec, script_name):
-    """Locate distribution `dist_spec` and run its `script_name` script"""
-    ns = sys._getframe(1).f_globals
-    name = ns['__name__']
-    ns.clear()
-    ns['__name__'] = name
-    require(dist_spec)[0].run_script(script_name, ns)
```

Comment:
Aah... I see the logic now: `_initialize_master_working_set()` is already doing this job.

I see why it can be removed now!

👍

Reply:
Yep exactly!

Given the questions about the other changes, I could split the removal of the unused `run_script`, `run_main`, `resource_dir`, since that's a safer bet.

Reply:
I've moved these changes to https://github.com/pypa/setuptools/pull/4260